src/storage/src/hummock/utils.rs

```diff
@@ -223,6 +233,16 @@ impl MemoryLimiter {
     }
 }
 
+impl MemoryTracker {
+    pub async fn increase_memory(&mut self, target: u64) {
```

Comment:
The method may potentially cause deadlock. For example, our total quota is 100, and currently each of two threads has obtained 40. And now when both of them want to increase 30 quota, deadlock may happen, since the remaining quota is 20, but both of them will not release quota before they are able to get the demanded 30 quota.

If this PR is urgent, we may have an issue to keep track of this potential problem and solve it in the future.

Reply:
Yes. But even if we do not support `increase_memory`, if two compaction task each hold one table, and they do not release the memory tracker unless the finished this task, they will cause deadlock when them acquire a new memory tracker for another table, which may be in another level of this task.